discussions/secure_web_api.md

```diff
@@ -0,0 +1,145 @@
+using tools like ```openssl```.
+
+This completely depends on the users, whether to use Trusted CA certificates or Self signed certificates. L3AF will not 
+provide any certificates.
+
+## L3AF deployment scenarios
+
+L3AF could be running in two scenarios, users can use L3AF in secure enterprise private networks and in public network.
+In case of private network, L3AFD and clients will be communicating with each other over a network that is normally
+protected by vpn or PCI (Payment Card Information), and hence it may not be essential to enable mTLS in this case.
+
+However, in case of public network, clients will be communicating with L3AFD over insecure networks. L3AF wants to
+secure its endpoints using industry's best standard available solutions. This can be configured from l3afd’s config file,
+and by default, mTLS will be disabled.
+
+## Enabling mTLS for L3AFD
+
+L3AFD (L3AF Daemon) will require a set of root certificates (root.crt and root.key) and a pair of server certificates
+(server.crt and server.key) to run in mTLS enabled mode. L3AFD will check for these certificates, and in case these are
+not found, L3AFD will stop with error ```certificates are not found```.
+
+The client will require a pair of client certificates (client.crt and client.key), generated from the same root
+certificates from which the server certificates have been generated, and the public root certificate (root.crt) to
+communicate with L3AFD.
+
+Process to enable mTLS 
+- Provision of Certificates 
+- Location of Certificates 
+- Enabling mTLS
+
+### Provision of Certificates
+
+As mentioned before, users can use already pre-existing certificates. However, L3AF can also provide mechanisms
+(manual or auto) to generate certificates.
+
+L3AF can have a library for self-signed certificate generation, that can implement APIs with different parameters, and
+these APIs can be integrated with L3AFD for auto-generation of the certificates. In this case, L3AFD can generate
+self-signed certificates, i.e., a set of self-signed root certificates, a pair of server certificates and client
+certificates at start-up if the certificates are not found (These certificates are not recommended for production use).
+
+Also, a certificate generation tool can be implemented using this library to generate the certificates manually.
+If the user has a set of root certificates (can be from a trusted CA provider or provided by its organisation or
+self-signed) and does not have the server and client certificates, then it can use the tool provided by L3AF to generate
+the server and client certificates. The tool can also have an option to generate the root certificates and the user can
+use this option to generate self-signed root certificates manually.
+
+#### Certificate creation options
+
+To create certificates L3AFD needs following options, these options are stored in L3AFD configuration file and for
+the tool, these can be passed as parameters.
```

Comment:
I'm assuming we only need to store certificate path(s) and cert password in the l3af config and not all the cert details?